### PR TITLE
Render conda-forge.yml docs from conda-smithy's JSON schema

### PR DESCRIPTION
**Pull request by jaimergp**
PR Checklist:

- [X] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [ ] if you are adding a new page under `docs/`, you have added it to the sidebar in `docs/sidebar.json`
- [ ] put any other relevant information below
- Part of #1971 
- Closes #2010. 
- Depends on https://github.com/conda-forge/conda-smithy/pull/1756.

This generates lightweight docs for `conda-forge.yml`. JSON Schema is fetched live. No need to rebuild the site if conda-smithy got updated.

Preview: https://deploy-preview-2095--conda-forge-previews.netlify.app/docs/maintainer/conda_forge_yml/

### CODE REVIEW

File: docs/maintainer/conda_forge_yml.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+The next section describes in detail the top-level fields in  `conda-forge.yml`.
+
+Note that each top-level CI provider field supports the `upload_packages` option.
+When set to False this will override the default behaviour of attempting to
+upload packages to anaconda.org, which can be useful for testing. For example:
+
+```yaml
+azure:
+  upload_packages: False
+```
```
Comment (jaimergp) [inline docs/maintainer/conda_forge_yml.mdx:23]: This should be part of the schema docs, not here.

**Comment by jaimergp** [comment]
With https://github.com/conda-forge/conda-smithy/pull/1756 merged, this is now pulling from conda-smithy @ main, so this is ready for a final review and merge.

File: .ci_scripts/update_docs [hunk 2]
```diff
@@ -12,9 +12,20 @@ if [[ "$CI" == "1" ]]; then
   git checkout -b new_site_content
 fi
 
+# Fetch conda-smithy JSON schema
+mkdir -p "$BASE_DIR/static/schema/"
+curl -sL \
+  -o "$BASE_DIR/static/schema/conda-forge.schema.json" \
+  "https://raw.githubusercontent.com/viniciusdc/conda-smithy/pydantic-schema-cf-yaml/conda_smithy/data/conda-forge.json"
+  # TODO^: replace with the real URL once the PR is merged
```
Comment (beckermr) [inline .ci_scripts/update_docs:20]: This needs to be addressed.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: .ci_scripts/update_docs (hunk 2 above)
Comment (jaimergp) [reply .ci_scripts/update_docs]: Oops, good catch! Thanks!

**Review by beckermr — APPROVED** [review]
Lgtm! This is great and thank you!

File: .ci_scripts/update_docs [hunk 3]
```diff
@@ -12,9 +12,19 @@ if [[ "$CI" == "1" ]]; then
   git checkout -b new_site_content
 fi
 
+# Fetch conda-smithy JSON schema
+mkdir -p "$BASE_DIR/static/schema/"
+curl -sL \
+  -o "$BASE_DIR/static/schema/conda-forge.schema.json" \
+  "https://raw.githubusercontent.com/conda-forge/conda-smithy/main/conda_smithy/data/conda-forge.json"
```
[the remark this answers is not in the document]
Comment (beckermr) [reply .ci_scripts/update_docs:19]: It should be from the latest release, but I discussed above with @jaimergp and we decided to punt on latest release for future work.

**Comment by h-vetinari** [comment]
This is very cool! 🤩 

Was just browsing a bit, but I think the dict for `os_version` is wrong (contains arches):
image

**Comment by jaimergp** [comment]
> Was just browsing a bit, but I think the dict for `os_version` is wrong (contains arches):

Do you mean it should just be `linux`? I think it does need the arch. [Examples in conda-forge](https://github.com/search?type=code&q=os_version+org%3Aconda-forge+path%3Aconda-forge.yml).

File: .ci_scripts/update_docs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaimergp) [reply .ci_scripts/update_docs:19]: I think the "vendored" version should be the released one (so people can use it as an autocomplete driver on their IDE), but then we fetch from `main` on every page load. However, there's no release now that features the schema, so we can change this when there's a new release.

[the remark this answers is not in the document]
Comment (jaimergp) [reply .ci_scripts/update_docs:19]: I'll merge and once there's a release we can update this line to `latest`. We might need an API call for that, or maybe push the file to GH Releases for simplicity 🤔

**Pull request merged**